lib/modules/platform/github/issue.ts

```diff
@@ -8,21 +9,132 @@ const GithubIssueBase = z.object({
   body: z.string(),
 });
 
+export interface GithubIssue {
+  body: string;
+  number: number;
+  state: string;
+  title: string;
+  lastModified: string;
+}
+
 const GithubGraphqlIssue = GithubIssueBase.extend({
   updatedAt: z.string(),
-}).transform((issue): Issue => {
+}).transform((issue): GithubIssue => {
   const lastModified = issue.updatedAt;
   const { number, state, title, body } = issue;
   return { number, state, title, body, lastModified };
 });
 
 const GithubRestIssue = GithubIssueBase.extend({
   updated_at: z.string(),
-}).transform((issue): Issue => {
+}).transform((issue): GithubIssue => {
   const lastModified = issue.updated_at;
   const { number, state, title, body } = issue;
   return { number, state, title, body, lastModified };
 });
 
 export const GithubIssue = z.union([GithubGraphqlIssue, GithubRestIssue]);
-export type GithubIssue = z.infer<typeof GithubIssue>;
+
+type CacheData = Record<number, GithubIssue>;
+
+export class GithubIssueCache {
+  private static reset(cacheData: CacheData): void {
+    delete this.issuesToReconcile;
+    const repoCache = getCache();
+    repoCache.platform ??= {};
+    repoCache.platform.github ??= {};
+    repoCache.platform.github.issuesCache = cacheData;
+  }
+
+  private static get data(): CacheData | undefined {
+    let cacheData = getCache().platform?.github?.issuesCache as
+      | CacheData
+      | undefined;
+    if (!cacheData) {
+      return undefined;
+    }
+
+    cacheData = this.reconcile(cacheData);
+
+    return cacheData;
+  }
+
+  static getIssues(): GithubIssue[] | null {
+    const cacheData = this.data;
+    if (!cacheData) {
+      return null;
+    }
+
+    const sortedResult = Object.values(cacheData).sort(
+      ({ lastModified: a }, { lastModified: b }) =>
+        DateTime.fromISO(b).toMillis() - DateTime.fromISO(a).toMillis(),
+    );
+
+    return sortedResult;
+  }
+
+  static setIssues(issues: GithubIssue[]): void {
+    const cacheData: CacheData = {};
+    for (const issue of issues) {
+      cacheData[issue.number] = issue;
+    }
+    this.reset(cacheData);
+  }
+
+  static updateIssue(issue: GithubIssue): void {
+    const cacheData = this.data;
+    if (cacheData) {
+      cacheData[issue.number] = issue;
+    }
+  }
+
+  private static issuesToReconcile: GithubIssue[] | undefined;
```

Comment:
UPD: this attempt wasn't successful

Reply:
I believe that the repository "fingerprint" concept was a defensive measure aimed to reduce chance of accidental data leakage, right? If so then maybe we could still intentionally work around it for the initial query because if the fingerprint is wrong then we can act accordingly immediately after?